scripts/lis_hyperv_platform/main-lis-rpm.ps1

```diff
@@ -0,0 +1,171 @@
+        [string] $VMSuffix,
+        [string] $KeyName
+    )
+    $xmlFullPath = Join-Path $PWD $Path
+    if (!(Test-Path $xmlFullPath)) {
+        throw "Test XML $xmlFullPath does not exist."
+    }
+    $xml = [xml](Get-Content $xmlFullPath)
+    $index = 0
+    if ($xml.config.VMs.vm -is [array]) {
+        foreach ($vmDef in $xml.config.VMs.vm) {
+            $xml.config.VMS.vm[$index].vmName = $VMSuffix
+            if ($KeyName) {
+                $xml.config.VMS.vm[$index].sshKey = $KeyName
+            }
+            $testParams = $vmDef.testParams
+            if ($testParams) {
+                $paramIndex = 0
+                foreach ($testParam in $testParams.param) {
+                    if ($testParam -like "VM2NAME=*") {
+                        $testParams.ChildNodes.Item($paramIndex)."#text" = `
+                            $testParam + $VMSuffix
+                    }
+                    $paramIndex = $paramIndex + 1
+                }
+            }
+            $index = $index + 1
+        }
+    } else {
+        $xml.config.VMS.vm.vmName = $VMSuffix
+        if ($KeyName) {
+            $xml.config.VMS.vm.sshKey = $KeyName
+        }
+    }
+    $xml.Save($xmlFullPath)
+}
+
+function Main {
+    if (!(Test-Path $WorkingDirectory)){
+        New-Item -ItemType directory -Path $WorkingDirectory
+    }
+    $jobPath = Join-Path -Path (Resolve-Path $WorkingDirectory) `
+        -ChildPath $JobId
+    New-Item -Path $jobPath -Type "Directory" -Force
+    $LISAPath = Join-Path $jobPath "lis-test"
+    $LISARelPath = Join-Path ${LISAPath} "WS2012R2\lisa"
+
+    Write-Host "Getting the proper VHD folder name for LISA with $DistroVersion"
+    $imageFolder = Join-Path $LISAImagesShareUrl $DistroVersion.split("_")[0]
+    $imageFolder = Join-Path $imageFolder $DistroVersion
+    
+    Write-Host "Getting LISA code..."
+    Get-LisaCode -LISAPath $LISAPath
+    
+    Write-Host "Copying lisa dependencies from share"
+    Copy-LisaTestDependencies `
+        -TestDependenciesFolders @("bin", "Infrastructure", "tools", "ssh") `
+        -LISARelPath $LISARelPath
+    pushd "${LISARelPath}\xml"
+    try {
+        Edit-TestXML -Path $XmlTest -VMSuffix $InstanceName
+    } catch {
+        throw
+    } finally {
+        popd
+    }
+
+    pushd $LISARelPath
+    Write-Host "Started running LISA"
+    try {
+        $lisaParams = ("LIS_URL='{0}';AZURE_TOKEN='{1}';LIS_URL_OLD='{2}'" -f @($LisUrl, $AzureToken, $LisOldUrl))
+        $lisaParams += ";username=lislabmsft;password=PA\$\$word;snapshotName=ICABase;DAEMONS=(kvp vss fcopy);HYPERV_MODULES=(hv_vmbus hyperv_keyboard hv_netvsc hid_hyperv hv_utils hv_storvsc hv_balloon)"
+        # Note(avladu): Lisa requires ErrorActionPreference = Continue,
+        # otherwise it will fail to run all the tests.
+        $ErrorActionPreference = "Continue"
+        if (!($LisaSuite)){
+            & .\lisa.ps1 -cmdVerb run -cmdNoun ".\xml\${XmlTest}" -dbgLevel 6 `
```

Comment:
use param expansion for creating the params, otherwise you end up with duplicate code

Reply:
example:

from: .\test.ps1  -test1 1 -test2 "test2"
you can do:
$a = @{"test1"=1; "test2"="test2"}
.\test.ps1 @a

Reply:
or if you have a boolean parameter (switch):
$a = @{"test2"="test2"; "test3"=2; "test1"=$true}
.\test.ps1 @a